freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UsersyncPoller.java

```diff
@@ -77,19 +81,25 @@ void syncFreeIpaStacks() {
                     .collect(Collectors.groupingBy(Stack::getAccountId))
                     .entrySet().stream()
                     .forEach(stringListEntry -> {
-                        UmsEventGenerationIds currentGeneration =
-                                umsEventGenerationIdsProvider.getEventGenerationIds(stringListEntry.getKey(), requestId);
-                        stringListEntry.getValue().stream()
-                                .forEach(stack -> {
-                                    if (isStale(stack, currentGeneration)) {
-                                        LOGGER.debug("Environment {} in Account {} is stale.", stack.getEnvironmentCrn(), stack.getAccountId());
-                                        SyncOperationStatus status = userService.synchronizeUsers(stack.getAccountId(), INTERNAL_ACTOR_CRN,
-                                                Set.of(stack.getEnvironmentCrn()), Set.of(), Set.of());
-                                        LOGGER.debug("Sync request resulted in operation {}", status);
-                                    } else {
-                                        LOGGER.debug("Environment {} in Account {} is up-to-date.", stack.getEnvironmentCrn(), stack.getAccountId());
-                                    }
-                                });
+                        String accountId = stringListEntry.getKey();
+                        if (entitlementService.automaticUsersyncPollerEnabled(INTERNAL_ACTOR_CRN, accountId)) {
```

Comment:
super nit: I would reverse the check and return if it's not entitled and gain one level of indentation.

Reply:
opened https://github.com/hortonworks/cloudbreak/pull/6610 to respond to this comment